src/adapter/bridges/OFTBridge.ts

```diff
@@ -0,0 +1,340 @@
+
+    assert(isDefined(l1Contract), `Cannot access L1 Bridge for token ${l1Token} when it is undefined.`);
+    return l1Contract;
+  }
+
+  /**
+   * This function is not supported. Inherited from BaseBridgeAdapter
+   */
+  protected getL2Bridge(): Contract {
+    throw new Error("Not supported for OFTBridge because we have multiple L2Bridge contracts");
+  }
+
+  /**
+   * Gets the L2 bridge contract for the specified token
+   * @param l1Token The L1 token address
+   * @returns The L2 bridge contract
+   */
+  protected getL2BridgeForL1Token(l1Token: string): Contract {
+    const l2TokenAddress = this.resolveL2TokenAddress(l1Token);
+    const l2Contract = this.oftContracts[l2TokenAddress]?.[this.l2chainId]?.contract;
+
+    assert(isDefined(l2Contract), `Cannot access L2 Bridge for token ${l2TokenAddress} when it is undefined.`);
+    return l2Contract;
+  }
+
+  // ! todo: check this ⚠️
+  async constructL1ToL2Txn(
+    toAddress: string,
+    l1Token: string,
+    _l2Token: string,
+    amount: BigNumber
+  ): Promise<BridgeTransactionDetails> {
+    // Verify the token is supported
+    assert(
+      OFTBridge.SUPPORTED_OFT_PATHS[l1Token]?.some((edge) => edge.l2ChainId === this.l2chainId),
+      `Token ${l1Token} is not supported for transfer from chain ${this.hubChainId} to ${this.l2chainId}`
+    );
+
+    // With `amount` here, we have to be mindful of `_removeDust` logic in `OFTCore` code
+    // TLDR: last `(supportedToken.decimals() - IOFT.sharedDecimals())` digits in amount have to be set to 0 to prevent rounding on the contract side
+    // more details here https://github.com/LayerZero-Labs/devtools/blob/a843edfc160f3cfa35d952b616b1cfe462503bc0/packages/oft-evm/contracts/OFTCore.sol#L343
+
+    const sendParamStruct: SendParamStruct = {
+      dstEid: this.l2DestinationEId,
+      to: oftAddressToBytes32(toAddress).toLowerCase(), // todo: .toLowerCase just in case, check this
+      amountLD: amount,
+      minAmountLD: amount,
+      extraOptions: "0x", // todo: 0x -- is this the way to do it?
+      composeMsg: "0x",
+      oftCmd: "0x",
+    };
+
+    // Get the L1 bridge contract for this token
+    const l1Bridge = this.getL1BridgeForToken(l1Token);
+
+    // todo: will this work?
+    const feeStruct: MessagingFeeStruct = await (l1Bridge as IOFT).quoteSend(sendParamStruct, false);
+
+    console.log("ethFee: ", feeStruct.nativeFee.toString());
+
+    // mult fee by 2 in order to get our tx included with high probability. Excess fee will get refunded
+    const ethFee = feeStruct[0].mul(2);
+
+    const newFeeStruct: MessagingFeeStruct = {
+      nativeFee: ethFee,
+      lzTokenFee: "0",
+    };
+
+    // ! note : if (msg.value != _nativeFee) revert NotEnoughNative(msg.value);
+    // ! nativeFee value has to be EXACTLY the msg.value
+
+    console.log("ethFee: ", ethFee.toString());
+
+    const cap = ethers.utils.parseEther("0.01");
+
+    if (ethFee.gt(cap)) {
+      throw "can't go over fee cap";
```

Comment:
```suggestion
      throw `can't go over fee cap (${ethFee} > ${cap})`;
```

Would be nice to include the amounts in the error here.